src/Microsoft.AspNetCore.Razor.Tools/Internal/PackageOnlyResolveTagHelpersRunCommand.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if NETCOREAPP1_0
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+using Microsoft.AspNetCore.Razor.Runtime.TagHelpers;
+using Microsoft.DotNet.InternalAbstractions;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.DotNet.ProjectModel.Loader;
+using Microsoft.Extensions.CommandLineUtils;
+using NuGet.Frameworks;
+
+namespace Microsoft.AspNetCore.Razor.Tools.Internal
+{
+    public class PackageOnlyResolveTagHelpersRunCommand : ResolveTagHelpersRunCommand
+    {
+        private readonly TagHelperTypeResolver packageTagHelperTypeResolver;
+
+        public PackageOnlyResolveTagHelpersRunCommand(AssemblyLoadContext loadContext)
+        {
+            packageTagHelperTypeResolver = new PackageTagHelperTypeResolver(loadContext);
+        }
+
+        protected override AssemblyTagHelperDescriptorResolver CreateDescriptorResolver() =>
+            new AssemblyTagHelperDescriptorResolver(packageTagHelperTypeResolver);
+
+        public static bool TryPackageOnlyTagHelperResolution(
```

Comment:
There's a parent try/catch in the `Program.Main`. Also, in the case that assembly load fails there's a try/catch mechanism built into `TagHelperTypeResolver` to gracefully handle those errors.